games-roguelike/dwarf-fortress/dwarf-fortress-0.47.05.ebuild

```diff
@@ -0,0 +1,84 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit optfeature toolchain-funcs
+
+MY_PV=$(ver_rs 0- _ "$(ver_cut 2-)")
+MY_PN=df
+MY_P=${MY_PN}_${MY_PV}
+
+DESCRIPTION="The deepest, most intricate simulation of a world that's ever been created."
+HOMEPAGE="https://www.bay12games.com/dwarves"
+SRC_URI="amd64? ( https://www.bay12games.com/dwarves/${MY_P}_linux.tar.bz2 )
+	x86? ( https://www.bay12games.com/dwarves/${MY_P}_linux32.tar.bz2 )"
+
+LICENSE="free-noncomm BSD BitstreamVera"
+SLOT="0"
+KEYWORDS="-* ~amd64 ~x86"
+IUSE="debug"
+
+RDEPEND="media-libs/glew:0
+	media-libs/libsdl[joystick,video]
+	media-libs/sdl-image[png]
+	media-libs/sdl-ttf
+	sys-libs/zlib
+	virtual/glu
+	x11-libs/gtk+:2"
+
+# Yup, libsndfile, openal and ncurses are only needed at compile-time; the code
+# dlopens them at runtime if requested.
+DEPEND="${RDEPEND}
+	media-libs/libsndfile
+	media-libs/openal
+	sys-libs/ncurses-compat:5[unicode]"
+BDEPEND="virtual/pkgconfig"
+
+S="${WORKDIR}/${MY_PN}_linux"
+
+gamesdir="/opt/${PN}"
+QA_PREBUILT="${gamesdir#/}/libs/Dwarf_Fortress"
+RESTRICT="strip"
+
+src_prepare() {
+	rm -f libs/*.so* || die
+	sed -i -e '1i#include <cmath>' g_src/ttf_manager.cpp || die
+	default
+}
+
+src_configure() {
+	tc-export CXX PKG_CONFIG
+	CXXFLAGS+=" -D$(use debug || echo N)DEBUG"
```

Comment:
append-cxxflags from flag-o-matic instead **but** this looks like it's actually a define, so it's CPPFLAGS anyway

Reply:
@thesamesam I understanding the different between `CPPFLAGS` and `CXXFLAGS` (the latter being a superset of the former).

I don't, however, understand what you mean by "looks like it's actually a define"

Would you care to elaborate on this a bit?

Reply:
Suggested to change `CXXFLAGS` to `append-cxxflags` but, given `-DNDEBUG` is "actually a define" (i.e. equivalent of `#define NDEBUG`), those would go in `CPPFLAGS` (C pre-processor, not cplusplus).

So, `append-cppflags` should be tried instead (if package respects it).

Reply:
That aside, I'm questioning why USE=debug exists here if this is all it does.

Reply:
@ionenwks thanks for your thoughts.

If you check the latest commit, [you'll notice](https://github.com/gentoo/gentoo/pull/20534/files#diff-960d65f2ff8959ba0548e979278eee2a8505e175ddf459502f467be84fbdc506R57) I've incorporated these changes.

I've tested, and `append-cppflags` works as expected. Also, I've grepped the code, and it is littered with `#if DEBUG`.

I opted to change the logic a bit - `-DNDEBUG` is a bit of a misnomer - on the one hand, it is benign, as there aren't any checks for it in the source code. On the other hand, I prefer to just selectively add `-DDEBUG`, and do nothing otherwise

Reply:
> I opted to change the logic a bit - -DNDEBUG is a bit of a misnomer - on the one hand, it is benign, as there aren't any checks for it in the source code.

There are a handful of assertions using the assert macro in the DF codebase, defining NDEBUG would make them not do anything.

Reply:
> > I opted to change the logic a bit - -DNDEBUG is a bit of a misnomer - on the one hand, it is benign, as there aren't any checks for it in the source code.
> 
> There are a handful of assertions using the assert macro in the DF codebase, defining NDEBUG would make them not do anything.

So you think i should add it back?

Can you point to a place where you think having `-DNDEBUG` would be helpful?

What about a compromise: `add-cppflags -DDEBUG=$(use debug)` (untested)

Reply:
I think this has been resolved. Please move review over to #20721